commit 8ab7f18123e67f4ee7c0888fa5b7157c482355ea
Author: Fabio M. Di Nitto <fabbione@fabbione.net>
Date:   Sun Apr 7 06:41:01 2024 +0200

    ipmi_sim_lan_control: read ip address from ipmi_sim config
    
    Signed-off-by: Fabio M. Di Nitto <fabbione@fabbione.net>

diff --git a/enginecore/ipmi_template/ipmi_sim_lancontrol b/enginecore/ipmi_template/ipmi_sim_lancontrol
index 368f29c..ae7418c 100755
--- a/enginecore/ipmi_template/ipmi_sim_lancontrol
+++ b/enginecore/ipmi_template/ipmi_sim_lancontrol
@@ -55,7 +55,8 @@ do_get() {
     while [ "x$1" != "x" ]; do
 	case $1 in
 	    ip_addr)
-		val=`ifconfig $device | grep 'inet' | awk '{print $2}'`
+		cfgfile="$(pstree -a $PPID | head -n 1 | sed -e 's#lan.conf.*#lan.conf#g' -e 's#.*/tmp#/tmp#g')"
+		val="$(cat $cfgfile |grep -v ".*#" | grep addr | awk '{print $2}')"
 		if [ "x$val" = "x" ]; then
 		    val="0.0.0.0"
 		fi
@@ -205,12 +206,11 @@ case $op in
     set)
 	do_set $@
 	;;
-    
     check)
 	do_check $@
 	;;
-
-*)
+    *)
 	echo "Unknown operation: $op"
 	exit 1
-esac
\ No newline at end of file
+	;;
+esac
